ee/codegen/src/generators/graph-attribute.ts

```diff
@@ -0,0 +1,409 @@
+import { python } from "@fern-api/python-ast";
+import { OperatorType } from "@fern-api/python-ast/OperatorType";
+import { AstNode } from "@fern-api/python-ast/core/AstNode";
+import { Writer } from "@fern-api/python-ast/core/Writer";
+
+import { PORTS_CLASS_NAME } from "src/constants";
+import { WorkflowContext } from "src/context";
+import { BaseNodeContext } from "src/context/node-context/base";
+import { PortContext } from "src/context/port-context";
+import { WorkflowDataNode, WorkflowEdge } from "src/types/vellum";
+
+// Fern's Python AST types are not mutable, so we need to define our own types
+// so that we can mutate the graph as we traverse through the edges.
+type GraphEmpty = { type: "empty" };
+type GraphSet = { type: "set"; values: GraphMutableAst[] };
+type GraphNodeReference = {
+  type: "node_reference";
+  reference: BaseNodeContext<WorkflowDataNode>;
+};
+type GraphPortReference = {
+  type: "port_reference";
+  reference: PortContext;
+};
+type GraphRightShift = {
+  type: "right_shift";
+  lhs: GraphMutableAst;
+  rhs: GraphMutableAst;
+};
+type GraphMutableAst =
+  | GraphEmpty
+  | GraphSet
+  | GraphNodeReference
+  | GraphPortReference
+  | GraphRightShift;
+
+export declare namespace GraphAttribute {
+  interface Args {
+    entrypointNodeEdges: WorkflowEdge[];
+    entrypointNodeId: string;
+    edgesByPortId: Map<string, WorkflowEdge[]>;
+    workflowContext: WorkflowContext;
+  }
+}
+
+export class GraphAttribute extends AstNode {
+  private readonly workflowContext: WorkflowContext;
+  private readonly edgesByPortId: Map<string, WorkflowEdge[]>;
+  private readonly entrypointNodeEdges: WorkflowEdge[];
+  private readonly entrypointNodeId: string;
+  private readonly astNode: python.AstNode;
+
+  public constructor({
+    entrypointNodeEdges,
+    entrypointNodeId,
+    edgesByPortId,
+    workflowContext,
+  }: GraphAttribute.Args) {
```

Comment:
I have ideas on how to simplify this constructor as we add more test cases. the issue is that we still reuse some of these entrypoint node and edges stuff for display class generation